res/controllers/mixxx-controls.d.ts

```diff
@@ -0,0 +1,439 @@
+declare type MixxxControlName = {
+  '[App]': MixxxAppControl;
+  '[Master]': MixxxMasterControl;
+  '[Microphone]': MixxxMicrophoneControl;
+  '[VinylControl]': MixxxVinylControl;
+  '[Recording]': MixxxRecordingControl;
+  '[AutoDJ]': MixxxAutoDJControl;
+  '[Library]': MixxxLibraryControl;
+  '[Shoutcast]': MixxxShoutcastControl;
+  '[Playlist]': MixxxPlaylistControl;
+  '[Controls]': MixxxControlsControl;
+  '[EffectRack1]': MixxxEffectEqualizerQuickEffectRack1Control;
+  '[EqualizerRack1]': MixxxEffectEqualizerQuickEffectRack1Control;
+  '[QuickEffectRack1]': MixxxEffectEqualizerQuickEffectRack1Control;
+  '[Skin]': MixxxSkinControl;
+} & {
+  [key: MixxxChannelGroup]: MixxxChannelControl;
+  [key: `[PreviewDeck${number}]`]: MixxxDeckControl;
+  [key: `[Sampler${number}]`]: MixxxSamplerControl;
+  [key: `[Microphone${number}]`]: MixxxMicrophoneControl;
+  [key: `[Auxiliary${number}]`]: MixxxAuxiliaryControl;
+  [key: `[EffectRack1_EffectUnit${number}]`]: MixxxEffectRack1UnitControl;
+  [
+    key: `[EqualizerRack1_[Channel${number}]]`
+  ]: EffectEqualizerQuickEffectRack1Control;
+  [
+    key: `[QuickEffectRack1_[Channel${number}]]`
+  ]: EffectEqualizerQuickEffectRack1Control;
+  [
+    key: `[EffectRack1_EffectUnit${number}_Effect${number}]`
+  ]: MixxxEffectEqualizerQuickEffectRack1EffectControl;
+  [
+    key: `[EqualizerRack1_[Channel${number}]_Effect1]`
+  ]: MixxxEffectEqualizerQuickEffectRack1EffectControl;
+  [
+    key: `[QuickEffectRack1_[Channel${number}]_Effect1]`
+  ]: MixxxEffectEqualizerQuickEffectRack1EffectControl;
+};
+
+type MixxxChannelGroup = `[Channel${number}]`;
+
+declare type MixxxGroup = keyof MixxxControlName;
+
+/*
+ * Values
+ */
+
+type MixxxPotMeterControls =
+  | '_up'
+  | '_down'
+  | '_up_small'
+  | '_down_small'
+  | '_set_one'
+  | '_set_minus_one'
+  | '_set_default'
+  | '_set_zero'
+  | '_toggle'
+  | '_minus_toggle';
+
+type MixxxAppControl =
+  | 'gui_tick_50ms_period_s'
+  | 'indicator_250ms'
+  | 'indicator_500ms'
+  | 'num_decks'
+  | 'num_samplers'
+  | 'num_preview_decks'
+  | 'num_microphones'
+  | 'num_auxiliaries'
+  | 'samplerate';
+
+type MixxxMasterControl =
+  | 'audio_latency_usage'
+  | 'audio_latency_overload'
+  | 'audio_latency_overload_count'
+  | 'balance'
+  | 'booth_enabled'
+  | 'booth_gain'
+  | 'crossfader'
+  | 'crossfader_down'
+  | 'crossfader_down_small'
+  | 'crossfader_up'
+  | 'crossfader_up_small'
+  | 'duckStrength'
+  | 'enabled'
+  | 'gain'
+  | 'headEnabled'
+  | 'headGain'
+  | 'headMix'
+  | 'headSplit'
+  | 'latency'
+  | 'num_effectsavailable'
+  | 'PeakIndicator'
+  | 'PeakIndicatorL'
+  | 'PeakIndicatorR'
```

Comment:
FYI these and VuMeter controls have been moved/renamed to the [Main] group.
https://github.com/mixxxdj/manual/pull/798